grafana-plugin/src/components/GForm/GForm.tsx

```diff
@@ -17,22 +17,57 @@ const nullNormalizer = (value: string) => {
   return value || null;
 };
 
-function renderFormControl(formItem: FormItem, register: any, control: any) {
+function renderFormControl(formItem: FormItem, register: any, control: any, onChangeFn: () => void) {
   switch (formItem.type) {
     case FormItemType.Input:
-      return <Input {...register(formItem.name, formItem.validation)} />;
+      return <Input {...register(formItem.name, formItem.validation)} onChange={onChangeFn} />;
 
     case FormItemType.TextArea:
-      return <TextArea rows={formItem.extra?.rows || 4} {...register(formItem.name, formItem.validation)} />;
+      return (
+        <TextArea
+          rows={formItem.extra?.rows || 4}
+          {...register(formItem.name, formItem.validation)}
+          onChange={onChangeFn}
+        />
+      );
+
+    case FormItemType.MultiSelect:
+      return (
+        <InputControl
+          render={({ field }) => {
+            return (
+              <GSelect
+                isMulti={true}
+                {...field}
+                {...formItem.extra}
+                onChange={(value) => {
+                  field.onChange(value);
+                  onChangeFn();
+                }}
+              />
+            );
+          }}
+          control={control}
+          name={formItem.name}
+        />
+      );
 
     case FormItemType.Select:
       return (
         <InputControl
           render={({ field: { onChange, ...field } }) => {
-            return <Select {...field} {...formItem.extra} onChange={(value) => onChange(value.value)} />;
+            return (
+              <Select
+                {...field}
+                {...formItem.extra}
+                onChange={(value) => {
+                  onChangeFn();
+                  onChange(value.value);
```

Comment:
can we change the name of value? to clearer view what is the value because value.value is not informative. But I am not sure what can it be, because it is a form